### PR TITLE
#522 - Create a GitHub templates for pull-requests and new issues

### PR DESCRIPTION
**Pull request by abstractj**
# Create a GitHub templates for pull-requests and new issues

## Summary 

This change resolves the issue #522 to introduce new templates for pull-requests, feature requests and bug reports.

## Type

[] Bug fix
[] Feature request
[x] Enhancement

## Why?

This is explained on issue #522

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+## Summary 
+
+<!-- 
+A quick summary about the feature request or bug fix with issue number. Ex: Add a new logging system described in #<issue number> 
+-->
+
+## Type
+
+[] Bug fix
+[] Feature request
```
Comment (JoelSpeed) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Should a feature request not be opened as an issue? Or are we planning to keep feature proposals somewhere in the repo before implementation?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+[] Enhancement
```
Comment (JoelSpeed) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Is it worth having a separate category for docs?

Comment (abstractj) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: @JoelSpeed updated now, please let me know if that works for you.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (abstractj) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: @JoelSpeed the proposal here is to be able to differentiate which pull requests are related to feature requests or bug fixes. So we can set the proper label and filter it.

This is how I see it:
- Feature requests: new features that people would like to see on Louketo. For example: Implement support to multiple upstreams.
- Enhancement: things like a typo, a small refactoring into the code. 

Maybe we can get rid of these labels and pick just one.


For feature proposals, something that will be a large feature and will change drastically the codebase, we can use [louketo-community](https://github.com/louketo/louketo-community) and create a folder "design-proposals". Does it make any sense?

But this is just an idea, we can change it, absolutely nothing is set in stone. Including the current labels, which I created to quickly migrate all the Jiras.

How does it work actually on the oauth2-proxy repository? If there's anything that you think should be changed, please let me know and we can adjust as we go.

**Comment by abstractj** [comment]
> Do we currently keep a changelog/are we planning to keep a changelog? It might be worth having a tickbox on the PR to remind authors to update the changelog if we are doing that?

@JoelSpeed do you mean something like https://github.com/oauth2-proxy/oauth2-proxy/blob/master/.github/PULL_REQUEST_TEMPLATE.md#checklist ?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (JoelSpeed) [reply .github/PULL_REQUEST_TEMPLATE.md:18]: That sounds perfectly reasonable to me, happy to keep this in. We can always iterate if we need to later.

Currently for categorising, normally the reviewer will look at it and mark it as the correct type. For proposals, we prefer to have new features discussed in an issue but this isn't always done and we haven't been strict about requiring it in the past. We actually discussed recently about changing the templates to tell people to create an issue to discuss changes before implementing them.

**Comment by JoelSpeed** [comment]
@abstractj That's exactly what I mean yes 😅 We use it as a hint to people that we expect them to keep the changelog and docs up to date in their PRs. Happy to have alternatives if you have a different way of doing this?

(The one about master branch is a leftover from some old CI process, it doesn't matter anymore and should be ditched)

**Comment by abstractj** [comment]
@JoelSpeed updated, please let me know if that's what you meant. In the future, we may consider to automate it with GH actions, in this way we don't need to worry about this.

**Comment by stianst** [comment]
> Do we currently keep a changelog/are we planning to keep a changelog? It might be worth having a tickbox on the PR to remind authors to update the changelog if we are doing that?

Changelog shouldn't be updated manually. Rather, issues should have milestones, which then allows folks to

**Comment by stianst** [comment]
Pressed wrong button, sorry ;)

**Comment by abstractj** [comment]
@stianst could you please review, so we can get it merged?

**Pull request merged**